components/MediList/index.js

```diff
@@ -18,36 +18,55 @@ export default function MediList() {
     setDosage('');
   };
 
+  const handleSearch = () => {
+    const filteredMedications = dummyMedications.filter(
+      (medication) =>
+        medication.medicationName.toLowerCase().includes(searchQuery.toLowerCase())
+    );
+
+    setMedications(filteredMedications);
+  };
+
   return (
     <StyledContainer>
+      <SearchContainer>
+        <input
+          type="text"
+          placeholder="Search for medications"
+          value={searchQuery}
+          onChange={(e) => setSearchQuery(e.target.value)}
+        />
+        <SearchButton onClick={handleSearch}>Search</SearchButton>
+      </SearchContainer>
+
       <Form onSubmit={handleAddMedikament}>
         <input
           type="text"
-          placeholder="Medikament eingeben"
+          placeholder="Enter Medication"
           value={medikament}
           onChange={(e) => setMedikament(e.target.value)}
           required
         />
         <DosageInputContainer>
           <input
             type="text"
-            placeholder="Dosierung eingeben"
+            placeholder="Enter Dosage"
             value={dosage}
             onChange={(e) => setDosage(e.target.value)}
             required
           />
-          <SubmitButton type="submit">Hinzufügen</SubmitButton>
+          <SubmitButton type="submit">Add Medication</SubmitButton>
         </DosageInputContainer>
       </Form>
 
-      <StyledContainer>
+      <MedicationListContainer>
         {dummyMedications.map((medication) => (
```

Comment:
You probably want to show all medications here (the state) instead of just the dummies.

Reply:
for now i just want to show dummies